crates/rollup-boost/src/server.rs

```diff
@@ -293,7 +303,7 @@ where
         builder_payload: OpExecutionPayloadEnvelope,
         payload_id: PayloadId,
         version: PayloadVersion,
-    ) -> RpcResult<Option<OpExecutionPayloadEnvelope>> {
+    ) -> BuilderResult {
```

Comment:
if builder_api_failed is always false, why not have the calling function do something like
```
            return self
                .calculate_external_state_root(payload, payload_id, version)
                .await
                .map(|payload| (payload, false));
```